plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -423,6 +431,15 @@ This typically means that we do a three-step process for a given definition:
 3. Modify the definition with the real body.
 -}
 
+{- Note [Occurrence analysis]
+GHC has "occurrence analysis", which is quite handy. In particular, it can tell you if variables are dead, which is useful
+in a couple of places.
+
+But it typically gets run *before* the simplifier, so when we get the expression we might be missing occurence analysis
+for any variables that were freshly created by the simplifier. That's easy to fix: we just run the occurrence analyser
+ourselves before we start.
+-}
+
```

Comment:
I think it might be better if the note would be closer to its referrer, in `plutus-tx-plugin/src/PlutusTx/Plugin.hs`.

Reply:
It is also referenced here!